terraform/eks/main.tf

```diff
@@ -81,47 +57,46 @@ provider "kubernetes" {
   load_config_file       = false
 }
 
-# create a unique namespace for each run
-resource "kubernetes_namespace" "aoc_ns" {
-  metadata {
-    name = "aoc-ns-${module.common.testing_id}"
-  }
-}
-resource "kubernetes_config_map" "aoc_config_map" {
-  metadata {
-    name      = "otel-config"
-    namespace = kubernetes_namespace.aoc_ns.metadata[0].name
-  }
-
-  data = {
-    "aoc-config.yml" = module.basic_components.otconfig_content
-  }
+provider "kubectl" {
+  // Note: copy from eks module. Please avoid use shorted-lived tokens when running locally.
+  // For more information: https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs#exec-plugins
+  host                   = data.aws_eks_cluster.testing_cluster.endpoint
+  cluster_ca_certificate = base64decode(data.aws_eks_cluster.testing_cluster.certificate_authority[0].data)
+  token                  = data.aws_eks_cluster_auth.testing_cluster.token
+  load_config_file       = false
 }
 
-# load eks pod config
-data "template_file" "eksconfig" {
-  template = file(local.eks_pod_config_path)
-
+data "template_file" "kubeconfig_file" {
+  template = file("./kubeconfig.tpl")
   vars = {
-    data_emitter_image = local.sample_app_image
-    testing_id         = module.common.testing_id
+    CA_DATA : data.aws_eks_cluster.testing_cluster.certificate_authority[0].data
+    SERVER_ENDPOINT : data.aws_eks_cluster.testing_cluster.endpoint
+    TOKEN = data.aws_eks_cluster_auth.testing_cluster.token
   }
 }
 
-# load the faked cert for mocked server
-resource "kubernetes_config_map" "mocked_server_cert" {
-  metadata {
-    name      = "mocked-server-cert"
-    namespace = kubernetes_namespace.aoc_ns.metadata[0].name
-  }
+resource "local_file" "kubeconfig" {
+  filename = "kubeconfig"
+  content  = data.template_file.kubeconfig_file.rendered
+}
 
-  data = {
-    "ca-bundle.crt" = module.basic_components.mocked_server_cert_content
+provider "helm" {
+  kubernetes {
+    host                   = data.aws_eks_cluster.testing_cluster.endpoint
+    cluster_ca_certificate = base64decode(data.aws_eks_cluster.testing_cluster.certificate_authority[0].data)
+    token                  = data.aws_eks_cluster_auth.testing_cluster.token
   }
 }
 
 locals {
-  eks_pod_config = yamldecode(data.template_file.eksconfig.rendered)["sample_app"]
+  eks_pod_config_path = fileexists("${var.testcase}/eks_pod_config.tpl") ? "${var.testcase}/eks_pod_config.tpl" : module.common.default_eks_pod_config_path
+}
+
+# create a unique namespace for each run
+resource "kubernetes_namespace" "aoc_ns" {
+  metadata {
+    name = "aoc-ns-${module.common.testing_id}"
```

Comment:
better have a var for namespace name as we use it multiplaces?

Reply:
Thanks for pointing out the issue. I don't think we should hardcode namespace elsewhere.
https://github.com/aws-observability/aws-otel-test-framework/pull/281#discussion_r624363642